### ISSUE TITLE
UX issue: backup/restore way too confusing, doesn't work half the time

### ISSUE THREAD

**Issue by ffxsam** [TablePlus/TablePlus#1088]
1. Which driver are you using and version of it (Ex: PostgreSQL 10.0): 10.5
 
2. Which TablePlus build number are you using (the number on the welcome screen, Ex: build 81): 212

3. The steps to reproduce this issue:

Backup from database on one machine, restore to another. All default options.

```
restore_pg_10.5: [archiver (db)] could not execute query: ERROR:  relation "adonis_schema" already exists
```

I've tried multiple variations of options.. clean, create, no-owner.. I simply cannot backup and restore a database at all. clean + create + no-owner seems the best, but it tries to drop the database (which is fine), except I get this error:

```
restore_pg_10.5: [archiver (db)] could not execute query: ERROR:  cannot drop the currently open database
    Command was: DROP DATABASE onspot_v2_dev;
```

Furthermore, it's very strange that I have to choose clean + create on the restore, even though the backup dump file includes instructions to drop the database and re-create everything.

Overall, the backup/restore process is extremely confusing and doesn't seem to work.

**Comment by huyphams** [comment]
The option single transaction will rollback all the changes if there is an error, so I think we can ignore the `relation "adonis_schema" already exists`, TablePlus uses pg_restore from PostgreSQL so it will generate the same error as restoring via command line.

By removing `single transaction` it will continue to restore and ignore the errors, you can review all the errors after restoring (mostly are duplicated user, grant tables... which can be ignored...).

<img width="195" alt="Screen Shot 2019-04-05 at 10 08 37 AM" src="https://user-images.githubusercontent.com/3654622/55601374-cb559e00-578a-11e9-86f9-6305fac6bdcf.png">

**Comment by ffxsam** [comment]
I'm not using `--single-transaction` in the examples above. Granted, it seems Postgres is a bit tricky with restores because you cannot drop the database you're restoring to. TablePlus should make this process a lot easier though.

**Comment by ianbeck** [comment]
Agreed, this is stupidly difficult. I need to dump a remote database, and then import table structure, constraints, and data locally in a different database (which may not be named the same). I have been attempting numerous different strategies in TablePlus and they have all been abysmal failures with difficult to understand error messages. Super frustrating. I tried Export/Import, but that doesn't retain constraints, evidently.

* PostgreSQL 10.10
* build 264

**Comment by JenCant** [comment]
Wasn't sure whether to create a new issue or to add my feedback here, but I agree with the general feedback on the usability of the backup/restore options.

After finally getting this process to work and successfully restoring backed up database, I've just discovered that it has wiped all the pre-existing users from the (local) server I just imported a database on, and have had to re-add and configure them all. 

I probably won't use this feature again unless I can feel a bit more confident over it's options and outcomes. It's a shame, because other than this and some minor niggles I've been really liking TablePlus. :) 

If it's any help, I've come to be using Tableplus from previously using Mysql workbench, Sequel Pro.

- Mariadb 10.1.37
- Build 272

**Comment by huyphams** [comment]
Hi guys, I'm sorry for any confusion caused. There are two different systems in TablePlus.
1. Backup/restore that uses the `embedded binaries` from the vendor MySQL, PostgreSQL. That means, it will fully compatible with MySQL and PostgreSQL.
However, it a bit tricky and hard to uses. You may want to read its documents in order to use it: MySQL: https://dev.mysql.com/doc/refman/8.0/en/mysqldump.html
PostgreSQL: https://www.postgresql.org/docs/12/app-pgdump.html

`TablePlus does not involve in this process, TablePlus only generate the connection string for backup/restore` that why we didn't have a progress bar.

2. The import/export: this is an implement of TablePlus, which includes CSV, JSON, and SQL. but it does not fully compatible with the database because the databases have many features that nearly impossible to catch up.

We're working on the next version of (2) the advanced import/export which will cover most common cases and its features can be compared with the tools you've mentioned above: SP, workbench etc...
